packages/block-editor/src/hooks/color.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { hasBlockSupport } from '@wordpress/blocks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getColorClassName,
+	getColorObjectByColorValue,
+	getColorObjectByAttributeValues,
+} from '../components/colors';
+import PanelColorSettings from '../components/panel-color-settings';
+import ContrastChecker from '../components/contrast-checker';
+import InspectorControls from '../components/inspector-controls';
+import { getBlockDOMNode } from '../utils/dom';
+
+export const COLOR_SUPPORT_KEY = '__experimentalColor';
+
+/**
+ * Filters registered block settings, extending attributes to include
+ * `backgroundColor` and `textColor` attribute.
+ *
+ * @param  {Object} settings Original block settings
+ * @return {Object}          Filtered block settings
+ */
+function addAttributes( settings ) {
+	if ( ! hasBlockSupport( settings, COLOR_SUPPORT_KEY ) ) {
+		return settings;
+	}
+
+	// allow blocks to specify their own attribute definition with default values if needed.
+	if ( ! settings.attributes?.backgroundColor ) {
+		settings.attributes = Object.assign( settings.attributes, {
+			backgroundColor: {
+				type: 'string',
+			},
+		} );
```

Comment:
Assigning the return value of [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) to the target is redundant. The function already mutates the target object. The code be optimized like so:
```suggestion
		Object.assign( settings.attributes, {
			backgroundColor: {
				type: 'string',
			},
		} );
```